src/main/java/Command.java

```diff
@@ -0,0 +1,213 @@
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        ui.printf("Here are the tasks in your list:\n" + tasks.toString());
+    }
+}
+
+/**
+ * A done command to mark a given task number as done
+ */
+class DoneCommand extends Command {
+    private int doneTask;
+
+    DoneCommand(int task) {
+        super();
+        this.doneTask = task;
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws IOException {
+        tasks.done(this.doneTask);
+        ui.printf("Nice! I've marked this task as done:\n" + tasks.get(this.doneTask));
+        storage.saveFile(tasks);
+    }
+}
+
+/**
+ * A delete command to delete a given task number
+ */
+class DeleteCommand extends Command {
+    private int deleteTask;
+
+    DeleteCommand(int task) {
+        super();
+        this.deleteTask = task;
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws IOException {
+        ui.printf("Noted. I've removed this task:\n" + tasks.get(this.deleteTask));
+        tasks.delete(this.deleteTask);
+        storage.saveFile(tasks);
+        ui.printf(tasks.taskCount());
+    }
+}
+
+/**
+ * A Todo command to add a todo Task to the TaskList
+ */
+class TodoCommand extends Command {
+    private String task;
+
+    TodoCommand(String toParse) {
+        this.task = toParse;
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws IOException {
+        Task task = tasks.addTodo(this.task);
+        ui.printf("Got it. I've added this task:\n" + task.toString());
+        ui.printf(tasks.taskCount());
+        storage.saveFile(tasks);
+    }
+}
+
+/**
+ * A Deadline command to add a Deadline Task to the TaskList
+ */
+class DeadlineCommand extends Command {
+    private String task;
+    private String deadline;
+
+    DeadlineCommand(String toParse) {
+        String[] split = toParse.split(" /by ");
+        this.task = split[0];
+        this.deadline = split[1];
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws IOException {
+        Task task = tasks.addDeadline(this.task, this.deadline);
+        ui.printf("Got it. I've added this task:\n" + task);
+        ui.printf(tasks.taskCount());
+        storage.saveFile(tasks);
+    }
+}
+
+/**
+ * An Event command to add an Event Task to the TaskList
+ */
+class EventCommand extends Command {
+    private String task;
+    private String at;
+    EventCommand(String toParse) {
+        String[] split = toParse.split(" /at ");
+        this.task = split[0];
+        this.at = split[1];
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws IOException {
+        Task task = tasks.addEvent(this.task, this.at);
+        ui.printf("Got it. I've added this task:\n" + task);
+        ui.printf(tasks.taskCount());
+        storage.saveFile(tasks);
+    }
+}
+
+class FindCommand extends Command {
```

Comment:
Shouldnt different classes be put into different files ?

Reply:
Agreed, for now the code is manageable as they are not too much, but if the project expands, it may become hard to navigate.